model/legacy-private/src/main/java/org/keycloak/migration/migrators/MigrateTo22_0_0.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.migration.migrators;
+
+import org.keycloak.migration.ModelVersion;
+import org.keycloak.models.ClientScopeModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.representations.idm.RealmRepresentation;
+
+public class MigrateTo22_0_0 implements Migration {
+
+    public static final ModelVersion VERSION = new ModelVersion("22.0.0");
+    public static final String OIDC_USERMODEL_PROPERTY_MAPPER = "oidc-usermodel-property-mapper";
+    public static final String SAML_USER_PROPERTY_MAPPER = "saml-user-property-mapper";
+    public static final String OPENID_CONNECT = "openid-connect";
+    public static final String SAML = "saml";
+    public static final String USER_ATTRIBUTE = "user.attribute";
+
+    @Override
+    public void migrate(KeycloakSession session) {
+        session.realms().getRealmsStream()
```

Comment:
Should we do the Liquibase migration for this instead? The advantage is that we don't need to load all realms and all client scopes, but we can update only those we are interested in. See for example this:
https://github.com/keycloak/keycloak/blob/main/model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/custom/JpaUpdate21_0_2_TermsAndConditionsRequiredAction.java

Reply:
The Liquibase migration turned out to be complicated as it would manipulate a CLOB field which needs different in the various databases. Liquibase's Java API also didn't allow for calculated values. 

The updated version of the code now handles old and new values, and old values print a warning.